src/Admin/AbstractAdmin.php

```diff
@@ -1099,8 +1099,7 @@ public function getActiveSubClass()
     {
         if (!$this->hasActiveSubClass()) {
             @trigger_error(sprintf(
-                'Calling %s() when there is no active subclass is deprecated since sonata-project/admin-bundle 3.52 and will throw an exception in 4.0. '.
-                'Use %s::hasActiveSubClass() to know if there is an active subclass.',
+                'Calling %s() when there is no active subclass is deprecated since sonata-project/admin-bundle 3.52 and will throw an exception in 4.0. Use %s::hasActiveSubClass() to know if there is an active subclass.',
```

Comment:
One of the goals of using `sprintf` is to avoid super long lines like this one… they make it harder to read the code IMO

Reply:
But the coding style of Symfony and yours indicates to declare exceptions in single line. See [example code](https://symfony.com/doc/current/contributing/code/standards.html) and [PHP CS Fixer rule](https://github.com/FriendsOfPHP/PHP-CS-Fixer).

> * **single_line_throw** [@Symfony]
>    Throwing exception must be done in single line.

In my projects, i do not approve of using single-line exceptions for the same reason. But you a public project that adheres to the Symfony coding style.

Reply:
Maybe we should amend CONTRIBUTING.md indeed, because this rule is bad, as you pointed out, and Yoda conditions are not great either. Maybe we should migrate to Doctrine CS?

Reply:
I can change this and limit the length of the message if it is approved by the miners (@sonata-project).

Reply:
Well it's more the line length that should be limited, the messages should be as long as necessary.

Reply:
> Well it's more the line length that should be limited, the messages should be as long as necessary.

Yes. That is exactly what i had in mind.

Reply:
With the [Doctrine CS](https://www.doctrine-project.org/projects/doctrine-coding-standard/en/latest/reference/index.html), not everything is so simple. Firstly, the Doctrine uses [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) and migrating to their CS requires changing the CS testing tool in order to always have the actual CS rules. Secondly, some of the Doctrine CS rules are also controversial:
> * Add spaces after a negation operator `if (! $cond)`
> * Add spaces around a colon in return type declaration `function () : void {}`

Reply:
Would be ok to me to ignore both of these 2 rules.

Reply:
And actually, Doctrine CS v8 drop the rule about the colon, see for instance https://github.com/doctrine/dbal/pull/4050

Reply:
I change usage throw to multiline